rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -145,19 +159,13 @@ class StaticMemoryStrategy : public memory_strategy::MemoryStrategy
   }
 
 private:
-  static const size_t pool_size_ = 1024;
-  static const size_t max_subscribers_ = 10;
-  static const size_t max_services_ = 5;
-  static const size_t max_clients_ = 10;
-  static const size_t max_guard_conditions_ = 50;
-  static const size_t max_executables_ = 1;
-
-  void * memory_pool_[pool_size_];
-  void * subscriber_pool_[max_subscribers_];
-  void * service_pool_[max_services_];
-  void * client_pool_[max_clients_];
-  void * guard_condition_pool_[max_guard_conditions_];
-  executor::AnyExecutable executable_pool_[max_executables_];
+  ObjectPoolBounds bounds_;
+
+  void ** memory_pool_;
+  void ** subscription_pool_;
+  void ** service_pool_;
+  void ** client_pool_;
```

Comment:
Wouldn't it simplify the code to use `std::vector` here instead of void pointers and custom allocation / deallocation?